Kernel/VM/PhysicalRegion.cpp

```diff
@@ -63,36 +64,74 @@ unsigned PhysicalRegion::finalize_capacity()
     return size();
 }
 
-RefPtr<PhysicalPage> PhysicalRegion::take_free_page(bool supervisor)
+Vector<RefPtr<PhysicalPage>> PhysicalRegion::take_contiguous_free_pages(size_t count, bool supervisor)
 {
     ASSERT(m_pages);
+    ASSERT(m_used != m_pages);
 
-    if (m_used == m_pages)
-        return nullptr;
+    Vector<RefPtr<PhysicalPage>> physical_pages;
+    physical_pages.ensure_capacity(count);
+
+    auto first_contiguous_page = find_contiguous_free_pages(count);
+
+    for (size_t index = 0; index < count; index++) {
+        physical_pages.append(PhysicalPage::create(m_lower.offset(PAGE_SIZE * (index + first_contiguous_page)), supervisor));
+    }
+    return physical_pages;
+}
 
+unsigned PhysicalRegion::find_contiguous_free_pages(size_t count)
+{
+    ASSERT(count != 0);
     // search from the last page we allocated
-    for (unsigned page = m_last; page < m_pages; page++) {
-        if (!m_bitmap.get(page)) {
-            m_bitmap.set(page, true);
-            m_used++;
-            m_last = page + 1;
-            return PhysicalPage::create(m_lower.offset(page * PAGE_SIZE), supervisor);
+    auto range = find_and_allocate_contiguous_range(m_last, m_pages, count);
+    if (!range.has_value()) {
+        // wrap back around to the start in case we missed something
+        range = find_and_allocate_contiguous_range(0, m_last, count);
+    }
+    ASSERT(range.has_value());
+    return range.value();
+}
+
+Optional<unsigned> PhysicalRegion::find_and_allocate_contiguous_range(unsigned first_page, unsigned last_page, size_t count)
```

Comment:
Can this function reuse what we already have in the `Bitmap` class, e.g. `Bitmap::find_longest_range_of_unset_bits`?